src/components/Modal.js

```diff
@@ -8,6 +8,8 @@ import SafeHTMLElement from '../helpers/safeHTMLElement';
 export const portalClassName = 'ReactModalPortal';
 export const bodyOpenClassName = 'ReactModal__Body--open';
 
+const useReactPortal = ReactDOM.createPortal !== undefined;
```

Comment:
Should we introduce a warning here indicating that usage without ReactDOM.createPortal is now deprecated and will be removed in the next major release?

Reply:
Then every React 15 user will get those warnings so they'll create issues about disabling those warnings.
Most likely some folks out there are using legacy React 14 even.

Reply:
@catamphetamine @nickhudkins Remember that all the changes to support react 16 must target the branch `next`. This way we can safely make any changes.

Reply:
What about upgrading to `3.0.0` and supporting only React 16+?

Reply:
@marlonbernardes It would be preferable to keep it backwards compatible. Also, to avoid maintaining 2 versions.

Reply:
It makes perfect sense. Let me know if there's anything I can help you with.